### PR TITLE
Fix bug and improve frontend client implemntation

### PR DESCRIPTION
**Pull request by georgolden**
- [x] tests and linter show no problems (`npm t`)
- [ ] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)
- [ ] description of changes is added in CHANGELOG.md

### CODE REVIEW

File: dist/metacom.js [hunk 1]
```diff
@@ -226,9 +214,8 @@ class HttpTransport extends Metacom {
       body: data,
     }).then((res) => {
       const { status } = res;
-      if (status === 200) {
+      if (status === 200 || status === 500) {
```
Comment (tshemsedinov) [inline dist/metacom.js:217]: Why just 500, do we need that for 404 for example?

File: dist/metacom.js [hunk 2]
```diff
@@ -19,21 +19,9 @@ class MetacomError extends Error {
...
-    const event = this._events.get(name);
-    if (event) event.add(fn);
-    else this._events.set(name, new Set([fn]));
-  }
-
-  emit(name, ...args) {
-    const event = this._events.get(name);
-    if (!event) return;
-    for (const fn of event.values()) fn(...args);
+    super();
```
Comment (tshemsedinov) [inline dist/metacom.js:24]: In this case we do not need constructor at all

Comment (georgolden) [reply dist/metacom.js:24]: MetacomInterface make sens for any features that may need in future

File: dist/metacom.js (hunk 1 above)
Comment (georgolden) [reply dist/metacom.js:217]: I tested and it is correct. There is no need for any additional status handling here. Just use `Metacom.message` it is more then enough.

File: dist/metacom.js (hunk 2 above)
Comment (georgolden) [reply dist/metacom.js:24]: I added a method for adding an Interface, if we need to have some Schemas scaffolding to a specific method, so we can just write any functionality for this with `MetacomInterface` in the same way, that we have in `impress` `Interfaces` and `Procedure` classes. There will be no need to make `Metacom.load` heavy.

**Comment by georgolden** [comment]
@tshemsedinov I added Semaphore usage fix to this PR. It is a small one fix, other Semaphore related issues will be made as separate PRs

**Comment by georgolden** [comment]
https://github.com/metarhia/metautil/pull/100 main fix for Semaphore

**Comment by georgolden** [comment]
Moved to https://github.com/metarhia/metacom/pull/295 and https://github.com/metarhia/metacom/pull/294

**Pull request closed without merging**